doc/manual/src/glossary.md

```diff
@@ -208,44 +208,85 @@
+    Otherwise, a [remote build](advanced-topics/distributed-builds.md) is needed.
 
-    A representation of the processor architecture and platform where build products will run on, typically a dash-separated string.
+  - [build platform]{#gloss-build-platform}
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    [build platform]: #gloss-build-platform
 
-    Programs that run natively in a [derivation] must be configured with a [host system] that can by executed by the [derivation system](language/derivations.md#attr-system).
+    The [platform] that performs the build or compilation, typically a dash-separated string.
 
-  - [target system]{#gloss-target-system}
+    A program that is produced by a [derivation] has a [build configuration] with a [build platform] that can be executed by the [derivation system](language/derivations.md#attr-system).
+    After it has been built, the [build platform] is no longer relevant.
 
-    [target system]: #gloss-target-system
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
 
-    A representation of the processor architecture and platform that a compiler will produce, typically a dash-separated string.
+  - [host platform]{#gloss-host-platform}
 
-    When discussing a build that does not produce a compiler, we only use [build system] and [host system] to describe it; we try to avoid discussing the actual build from the perspective its compiler's configuration.
+    [host platform]: #gloss-host-platform
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    The [platform] where build products will run on, typically a dash-separated string.
 
-    If the output of a compiler is to be run in a [derivation], the compiler's [target system] must be executable by the [derivation system](language/derivations.md#attr-system).
+    Programs that run natively in a [derivation] must have a [build configuration] with a [host platform] that can by executed by the [derivation system](language/derivations.md#attr-system).
+
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+
+  - [target platform]{#gloss-target-platform}
+
+    [target platform]: #gloss-target-platform
+
+    The [platform] for which a compiler will generate code, typically a dash-separated string.
+
+    When discussing a build that does not produce a compiler, we only use [build platform] and [host platform] to describe it. This also applies to compilers with a dynamic target, which is not part of the [build configuration].
+
+    We try to avoid discussing the actual build from the perspective of its compiler, instead focusing on the build itself.
+    Otherwise we won't have a term to distinguish the actual [target platform] from what we should have called the [host platform].
```

Comment:
```suggestion
```

I don't understand what this is trying to tell me.

Reply:
A lot of compiler writers screwed this up, because they only thought of the terms from *their* perspective and not *their user's* perspective.